neo4j/internal/pool/pool.go

```diff
@@ -343,16 +343,18 @@ func (p *Pool) tryBorrow(
 	return c, nil
 }
 
-func (p *Pool) unreg(ctx context.Context, serverName string, c idb.Connection, now time.Time) {
+func (p *Pool) unreg(ctx context.Context, serverName string, c idb.Connection, now time.Time, close bool) {
 	p.serversMut.Lock()
 	defer p.serversMut.Unlock()
-	p.unregLocked(ctx, serverName, c, now)
+	p.unregLocked(ctx, serverName, c, now, close)
 }
 
-func (p *Pool) unregLocked(ctx context.Context, serverName string, c idb.Connection, now time.Time) {
+func (p *Pool) unregLocked(ctx context.Context, serverName string, c idb.Connection, now time.Time, close bool) {
 	defer func() {
 		// Close connection in another thread to avoid potential long blocking operation during close.
-		go c.Close(ctx)
+		if close {
```

Comment:
Is this to do with the threading github issue? If so might be worth mentioning it in the PR description

Reply:
Yeah good spot - this is what highlighted the issue, running the close in another thread. Turns out the issue was the logic in the pool return function which was fixed in: https://github.com/neo4j/neo4j-go-driver/pull/610. 👍